src/rpc/server.cpp

```diff
@@ -434,6 +433,47 @@ static inline JSONRPCRequest transformNamedArguments(const JSONRPCRequest& in, c
     return out;
 }
 
+/**
+ * Process positional arguments into an Object of named arguments, based on the
+ * passed-in specification for the RPC call's arguments.
+ */
+static inline UniValue transformPositionalArguments(const UniValue& in, const std::vector<std::string>& argNames)
+{
+    assert(in.size() <= argNames.size());
+    UniValue out(UniValue::VOBJ);
+    for (size_t i = 0; i < in.size(); ++i) {
+        const UniValue &val = in[i];
+        const std::string &argNamePattern = argNames[i];
+        std::vector<std::string> vargNames;
+        boost::algorithm::split(vargNames, argNamePattern, boost::algorithm::is_any_of("|"));
+        for (const std::string & argName : vargNames) {
+            out.pushKV(argName, val);
+        }
+    }
+    // Return request with positional arguments transformed to named arguments
+    return out;
+}
+
+static inline JSONRPCRequest transformArguments(const JSONRPCRequest& in, const std::vector<std::string>& argNames)
+{
+    auto out = in;
+    if (in.params.isObject()) {
```

Comment:
Nit: wouldn't it be slightly cleaner to do something like:
```
bool isPositional = in.params.isObject();
out.m_used_positional_args = isPositional;
if (isPositional) { ... }
```

Same loc count, but reads slightly cleaner IMHO.